recipes-extended/realsense-camera/realsense-camera/0001-librealsense-Fix-link-when-the-system-library-is-use.patch

```diff
@@ -0,0 +1,30 @@
+From fb74e5eec2368f20c4b43cf9224519b29aee35ad Mon Sep 17 00:00:00 2001
+From: Murilo Belluzzo <murilo.belluzzo@intel.com>
+Date: Wed, 5 Apr 2017 17:06:44 -0700
+Subject: [PATCH 3/3] librealsense: Fix link when the system library is used
+
+Otherwise librealsense won't be listed as a dependency for the nodelet
+that will fail to load due to undefined symbols.
+
+Signed-off-by: Murilo Belluzzo <murilo.belluzzo@intel.com>
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a1ec2f6..926992a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -106,6 +106,12 @@ add_library(${PROJECT_NAME}_nodelet src/base_nodelet.cpp src/r200_nodelet.cpp sr
+ target_link_libraries(${PROJECT_NAME}_nodelet
+   ${catkin_LIBRARIES}
+ )
++if(USE_SYSTEM_LIBREALSENSE)
++  target_link_libraries(${PROJECT_NAME}_nodelet
++    ${realsense_LIBRARY}
```

Comment:
I care because is the only working option for most of the projects relying on librealsense stable versions up to 1.12.1.

If you take a look at your first reference, it states the above (in the plural form, to be honest, that is what I actually find on all custom FindRealsense.cmake I have to deal frequently, besides other packages also returning on this form).

[2] and [3] make use of this horrible idea without any good justification. But the fact is that it isn't widely used. Just grep all .cmake files on work dir of a a yocto build and see.

The config should return the namespace form indeed, but the user should rely on the "expected" output (I say "expected" because oficial documentation of this function is terrible like the overall cmake documentation usually is, although having improved a lot along the years) of find_package() as stated in [1] and not try to guess the value. 

And last but not least, if wasn't a common practice, why would you care to set the value of <package>_LIBRARY (going against your own references) as the last step of the cmake config that you wrote for librealsense?  :-/.

This complains makes no sense to me, it should actually be s/realsense_LIBRARY/realsense_LIBRARIES/ and a patch on your config upstream.